server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -270,6 +271,14 @@ public final String param(String key, String defaultValue) {
         return params;
     }
 
+    public void params(Map<String, String> params) {
+        for (Map.Entry<String, String> entry : params.entrySet()) {
+            if(this.params.get(entry.getKey()) == null) {
+                this.params.put(entry.getKey(), entry.getValue());
+            }
+        }
+    }
```

Comment:
I was hoping that we could have shared code between search and msearch to keep the supported params in sync, without having to change RestRequest. Do you think that is possible?